test.py

```diff
@@ -0,0 +1,97 @@
+import pytest
+from menu_delivery import menu_delivery
+from PyInquirer import Validator, ValidationError
+
+
+class document:
+    """Sample class for temporary document object"""
+
+    def __init__(self, text):
+        """creates a text element
+
+        Args:
+            text (string): test string for any document object
+        """
+        self.text = text
+
+
+def test_ip_standard():
+    """Tests standard ip addresses"""
+    IPValidator = menu_delivery.IPValidator()
+    test_ip_list = [
+        document("127.0.0.1"),
+        document("192.168.0.1"),
+        document("172.168.1.4"),
+        document("10.0.0.1"),
+    ]
+    for test_ip in test_ip_list:
+        try:
+            IPValidator.validate(test_ip)
+        except Exception as exc:
+            assert False
+
+
+def test_ip_string():
+    """Tests string in IP"""
+    IPValidator = menu_delivery.IPValidator()
+    with pytest.raises(ValidationError):
+        IPValidator.validate(document("asgsadfg"))
+
+
+def test_ip_two_dots():
+    """Tests incomplete IP"""
+    IPValidator = menu_delivery.IPValidator()
+    with pytest.raises(ValidationError):
+        IPValidator.validate(document("127.0.0"))
+
+
+def test_ip_one_dot():
+    """Tests incomplete IP"""
+    IPValidator = menu_delivery.IPValidator()
+    with pytest.raises(ValidationError):
+        IPValidator.validate(document("127.0"))
+
+
+def test_ip_whole_number():
+    """Tests incomplete IP"""
+    IPValidator = menu_delivery.IPValidator()
+    with pytest.raises(ValidationError):
+        IPValidator.validate(document("127"))
+
+
+def test_ip_special_character():
+    """Tests special characters in ip"""
+    IPValidator = menu_delivery.IPValidator()
+    with pytest.raises(ValidationError):
+        IPValidator.validate(document("##.##.##.##"))
```

Comment:
are there any other invalid ips u can think of? any special cases that might break it? (something like 11111111.1.1.1 maybe)